### PR TITLE
sample defaults

### PR DESCRIPTION
**Pull request by chexov**

### CODE REVIEW

File: src/main/java/com/vg/live/SuperLive.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ public static boolean isEOF(ByteBuffer buf) {
         int sampleFlags = tfhd.getDefaultSampleFlags();
 
         for (int i = 0; i < sampleCount; i++) {
-            if (!trun.isFirstSampleFlagsAvailable()) {
+            if (trun.getSamplesFlags() != null) {
```
Comment (chugai) [inline src/main/java/com/vg/live/SuperLive.java]: what if sample sizes are explicitly set in trun? would you read them if trun.getSamplesFlags() == null ?

Comment (chexov) [reply src/main/java/com/vg/live/SuperLive.java]: @chugai sizes, flags, and duration are in the same table.
And I am assuming that If samplesFlags[] == null then sampleSizes and sampleDurations arrays are null

Comment (zhuker) [reply src/main/java/com/vg/live/SuperLive.java]: > And I am assuming that If samplesFlags[] == null then sampleSizes and sampleDurations arrays are null

not a valid assumption

Comment (chugai) [reply src/main/java/com/vg/live/SuperLive.java]: and first sample flag support?

**Comment by chexov** [comment]
image

File: src/test/java/com/vg/live/SuperLiveTest.java [hunk 2]
```diff
@@ -9,25 +9,50 @@
...
-        File init = new File("testdata/trun/stream-0.0.m4s");
-        File m4s = new File("testdata/trun/stream-0.42.m4s");
+    public void sampleCount() {
+        File init = new File(M4S_INIT);
+        File m4s = new File(M4S);
         File file = new File("testdata/trun/stream-0.116.m4s");
 
-        Observable<ByteBuffer> bufs = just(init, m4s, file).map(f1(f -> wrap(readFileToByteArray(f))));
+        Observable<ByteBuffer> bufs = just(init, m4s, file).map(f1(f -> {
+            System.out.println("f=" + f.getAbsolutePath());
```
Comment (zhuker) [inline src/test/java/com/vg/live/SuperLiveTest.java:31]:
```
.doOnNext(f -> sysout(f))
.map(f -> wrap(readFileToByteArray(f))
```

**Review by zhuker — DISMISSED** [review]

**Pull request merged**